CRM/Upgrade/Incremental/php/FiveEleven.php

```diff
@@ -67,21 +67,14 @@ public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
    * (change the x in the function name):
    */
 
-  //  /**
-  //   * Upgrade function.
-  //   *
-  //   * @param string $rev
-  //   */
-  //  public function upgrade_5_0_x($rev) {
-  //    $this->addTask(ts('Upgrade DB to %1: SQL', array(1 => $rev)), 'runSql', $rev);
-  //    $this->addTask('Do the foo change', 'taskFoo', ...);
-  //    // Additional tasks here...
-  //    // Note: do not use ts() in the addTask description because it adds unnecessary strings to transifex.
-  //    // The above is an exception because 'Upgrade DB to %1: SQL' is generic & reusable.
-  //  }
-
-  // public static function taskFoo(CRM_Queue_TaskContext $ctx, ...) {
-  //   return TRUE;
-  // }
+  /**
+   * Upgrade function.
+   *
+   * @param string $rev
+   */
+  public function upgrade_5_11_alpha1($rev) {
+    $this->addTask(ts('Upgrade DB to %1: SQL', array(1 => $rev)), 'runSql', $rev);
+    $this->addTask('Update smart groups where jcalendar fields have been converted to datepicker', 'updateSmartGroups', $rev);
+  }
```

Comment:
This is excellent work @eileenmcnaughton. I have one nitpick about the design. The way you've done it here, the `addTask` function passes a version number to `updateSmartGroups` which then relies on the data in `getSmartGroupConversions` to figure out what to do. That's a bit hard to grok and seems like it might cause trouble for future devs trying to figure out and extend what you've done. Since the `addTask` function can accept a variable number of arguments, I'd suggest passing in whatever's needed directly, e.g.

        $this->addTask('Update grant smart groups for datepicker', 'updateSmartGroups', $rev, [
          'function' => 'datepickerConversion',
          'fields' => [
            'grant_application_received_date',
            'grant_decision_date',
            'grant_money_transfer_date',
            'grant_due_date'
          ]
        ]);